src/cli/allow_ip.rs

```diff
@@ -0,0 +1,147 @@
+use crate::cli::util::{cluster_from_conn_str, find_org_id, find_project_id};
+use crate::cli::{client_error_to_shell_error, no_active_cluster_error};
+use crate::state::State;
+use nu_engine::CallExt;
+use nu_protocol::ast::Call;
+use nu_protocol::engine::{Command, EngineState, Stack};
+use nu_protocol::{Category, PipelineData, ShellError, Signature, SyntaxShape, Value};
+use std::ops::Add;
+use std::sync::{Arc, Mutex};
+use tokio::time::Instant;
+
+#[derive(Clone)]
+pub struct AllowIP {
+    state: Arc<Mutex<State>>,
+}
+
+impl crate::cli::AllowIP {
+    pub fn new(state: Arc<Mutex<State>>) -> Self {
+        Self { state }
+    }
+}
+
+impl Command for crate::cli::AllowIP {
+    fn name(&self) -> &str {
+        "allow ip"
+    }
+
+    fn signature(&self) -> Signature {
+        Signature::build("allow ip")
+            .category(Category::Custom("couchbase".to_string()))
+            .optional(
+                "address",
+                SyntaxShape::String,
+                "ip address to allow access to the cluster",
+            )
+    }
+
+    fn usage(&self) -> &str {
+        "Whitelists an IP address on a Capella cluster"
```

Comment:
"Adds an IP address to the allowlist on a Capella cluster" maybe? I think we've moved away from using whitelist?

Reply:
Done!